### PR TITLE
Enable geo_distance and geo_bounding_box queries on geo_shape field type

### LINKED ISSUE #54628: Enable geo_distance and geo_bounding_box queries on geo_shape field type
Following the development in #48928, we should enable `geo_distance` and `geo_bounding_box` queries over `geo_shape` field type.

### PR DESCRIPTION
**Pull request by iverase**
This commit enables geo_distance and geo_bounding_box queries on geo_shape field type. In addition it enables those queries over geo_point runtime fields as well.

closes #54628

### CODE REVIEW

File: docs/reference/query-dsl/geo-bounding-box-query.asciidoc [hunk 1]
```diff
@@ -316,6 +395,8 @@ that the `geo_point` type must have lat and lon indexed in this case).
 Note, when using the indexed option, multi locations per document field
 are not supported. Here is an example:
 
+//./gradlew ':docs:integTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/query-dsl/geo-bounding-box-query/*}"
```
Comment (talevy) [inline docs/reference/query-dsl/geo-bounding-box-query.asciidoc:398]: whoa, I didn't know this is possible

**Review by talevy — APPROVED** [review]
Overall looks great! I read the docs, the content looks good. There are some English things that @jrodewig could comment on maybe?

File: docs/reference/query-dsl/geo-distance-query.asciidoc [hunk 2]
```diff
@@ -64,6 +89,57 @@ GET /my_locations/_search
...
+            "lon": -70
+          }
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+Or it can be executed in both indexes at the same time:
```
Comment (jrodewig) [inline docs/reference/query-dsl/geo-distance-query.asciidoc:117]: Current text is fine. Feel free to ignore if wanted.
```suggestion
To match both `geo_point` and `geo_shape` values, search both indices:
```

**Review by jrodewig — CHANGES_REQUESTED** [review]
This looks good overall. However, there are some grammar issues that need to be fixed before a merge. It also looks like a few of the example snippets target the wrong indices.

**Comment by iverase** [comment]
Thanks you @jrodewig, I added your suggestions!

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @iverase!

**Pull request merged**